doc/nrf/config_and_build/bootloaders_and_dfu/bootloader_adding.rst

```diff
@@ -286,35 +296,41 @@ The following section describes how to add it to your secure bootloader chain.
 Adding MCUboot as an upgradable bootloader
 ==========================================
 
-To use MCUboot as an upgradable bootloader, the application must already use the |NSIB| as the immutable bootloader.
-MCUboot is added to its boot chain by including the :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` Kconfig option with either the build command or in the application's :file:`prj.conf` file:
+To use MCUboot as an upgradable bootloader to your application, complete the following steps:
 
-.. code-block::
+1. :ref:`Add |NSIB| as the immutable bootloader <ug_bootloader_adding_immutable_b0>`.
+#. Define :ref:`static partition for the immutable bootloader <ug_pm_static>`.
+   This is important because adding an upgradable bootloader to an application changes its :ref:`flash memory partition layout <ug_bootloader_flash_mcuboot>` to accommodate first and second image slots for both immutable and upgradable bootloaders.
```

Comment:
You do not have to use a static partition for developement, it becomes important for production.
Also, the first build you do, you can use dynamic partitioning. As long as all subsequent builds use the same partitioning as was dynamically created.
Dynamic partitioning is so much easier that I  generally recommend starting by doing this, to see how the system generated the partitioning. Then you can create static partitioning from what you learned.
Can I suggest that we mention static partition later on in this list instead?

Reply:
Maybe we could mention that user can play around with dynamic partitions as long as DFU is not used. We need to keep in mind that the dynamic partitions may not be backwards compatible and upgrading firmware may lead to strange issues when dynamic partition maps are not interoperable.

Reply:
> Maybe we could mention that user can play around with dynamic partitions as long as DFU is not used.

I think this one points to the important difference between DFU and bootloaders.
And as you say, this limitation is only relevant for DFU.
Therefore, it could be argued that we do not need to mention static partitioning here, but leave it for a "how to enable DFU" section? 
However, I think I like it being mentioned here still.

Reply:
In theory yes. But I wonder if there are practical use-cases of using bootloader without DFU mechanism (apart from simple local checks on desk). That's why I was wondering about simplifying this part.

Reply:
Let's keep this info next to where we mention flash memory partitions.

Reply:
> Do we have use-cases where bootloader without DFU mechanism is used?
Not the most widely used, but for devices without a KMU, the [HUK ](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/keys/hw_unique_key/README.html#overview )is protected by a bootloader before the application runs:
However, [PSA Certified Level 1](https://www.psacertified.org/app/uploads/2022/06/JSADEN001-PSA_Certified_Level_1-2.2-REL-01.pdf) requires upgradability, so using HUK without PSA Certifiication seems quite niche.